File: android/app/src/main/java/com/nhs/online/nhsonline/registration/PaycassoCallbackResponse.kt
package com.nhs.online.nhsonline.registration

import com.google.gson.annotations.SerializedName

data class PaycassoCallbackResponse @JvmOverloads constructor(
    @SerializedName("transactionId") val transactionId: String? = null,
    @SerializedName("transactionType") val transactionType: String? = null,
    @SerializedName("isFaceMatched") val isFaceMatched: Boolean? = false,
    @SerializedName("error") val paycassoError: PaycassoError?
)

data class PaycassoError @JvmOverloads constructor(
    @SerializedName("errorCode") val errorCode: Int? = null,
    @SerializedName("errorMessage") val errorMessage: String)

File: xamarin/NHSOnline.App.iOS/DependencyServices/PaycassoCallbackResponse.cs
namespace NHSOnline.App.iOS.DependencyServices
{
    public sealed class PaycassoCallbackResponse
    {
        private PaycassoCallbackResponse()
        { }

        public string? TransactionId { get; set; }
        public PaycassoTransactionType? TransactionType { get; set; }
        public bool? IsFaceMatched { get; set; } = false;
        public PaycassoError? Error { get; set; }

        public static PaycassoCallbackResponse ForSuccess(
            string transactionId,
            PaycassoTransactionType transactionType,
            bool isFaceMatched,
            PaycassoError? error)
        {
            return new PaycassoCallbackResponse
            {
                TransactionId = transactionId,
                TransactionType = transactionType,
                IsFaceMatched = isFaceMatched,
                Error = error
            };
        }

        public static PaycassoCallbackResponse ForError(string errorMessage, int? errorCode = null)
        {
            return new PaycassoCallbackResponse
            {
                Error = new PaycassoError(errorMessage)
                {
                    ErrorCode = errorCode
                }
            };
        }
    }
}